packages/bitcore-lib/lib/crypto/signature.js

```diff
@@ -26,9 +27,11 @@ Signature.prototype.set = function(obj) {
   this.r = obj.r || this.r || undefined;
   this.s = obj.s || this.s || undefined;
 
-  this.i = typeof obj.i !== 'undefined' ? obj.i : this.i; //public key recovery parameter in range [0, 3]
-  this.compressed = typeof obj.compressed !== 'undefined' ?
-    obj.compressed : this.compressed; //whether the recovered pubkey is compressed
+  // public key recovery parameter in range [0, 3]
+  this.i = typeof obj.i !== 'undefined' ? obj.i : this.i;
```

Comment:
Recommend switching from "If not A then B, else C" to "If A then C, else B"
`this.i = typeof obj.i === 'undefined' ? this.i : obj.i`

Same as below